src/OpenTelemetry/Metrics/CompositeMetricReader.cs

```diff
@@ -57,12 +60,117 @@ public CompositeMetricReader AddReader(MetricReader reader)
             };
             this.tail.Next = node;
             this.tail = node;
+            this.count++;
 
             return this;
         }
 
         public Enumerator GetEnumerator() => new Enumerator(this.head);
 
+        internal List<Metric> AddMetricsWithNoViews(Instrument instrument)
+        {
+            var metrics = new List<Metric>();
+            for (var cur = this.head; cur != null; cur = cur.Next)
+            {
+                var metric = cur.Value.AddMetricWithNoViews(instrument);
+                metrics.Add(metric);
```

Comment:
add only if notnull

Reply:
and in the provider, you can check the list.Count > 0 before enable measurement event is called.

Reply:
I'm adding `null` to preserve the order in which we process the metrics when we have multiple readers. `CompositeMetricReader` checks for `null` before calling the reader to update the value.

In future, if we allow readers to have their own `MaxMetrics` limit, this would still work as we will know which readers need to update the value and which don't.